server/src/main/java/org/opensearch/search/SearchService.java

```diff
@@ -389,6 +400,9 @@ public SearchService(
 
         lowLevelCancellation = LOW_LEVEL_CANCELLATION_SETTING.get(settings);
         clusterService.getClusterSettings().addSettingsUpdateConsumer(LOW_LEVEL_CANCELLATION_SETTING, this::setLowLevelCancellation);
+
+        IndexSearcher.setMaxClauseCount(INDICES_MAX_CLAUSE_COUNT_SETTING.get(settings));
+        clusterService.getClusterSettings().addSettingsUpdateConsumer(INDICES_MAX_CLAUSE_COUNT_SETTING, IndexSearcher::setMaxClauseCount);
```

Comment:
Can we add local settings consumer? That can further delegate the new value to IndexSearcher.setMaxClauseCount. The local attribute can be used as is within `checkForTooManyFields` method above.

Reply:
thx @jainankitk for the feedback! I'm wondering what the use-case for the consumer might be? If we're only setting the searcher clause count and feeding it to the settings consumer wouldn't that be enough?

Reply:
What is the purpose of below expression? Is it possible to override this setting at index or request level? If not, we can add getter in search using the consumer mentioned above.

```
Integer limit = SearchService.INDICES_MAX_CLAUSE_COUNT_SETTING.get(context.getIndexSettings().getSettings());
```

Reply:
It registers a consumer for and adds it to the cluster settings. The consumer is the `IndexSearcher.setMaxClauseCount`.

> Is it possible to override this setting at index or request level?

It is only possible to override this setting at the cluster level since we'd want all indices to have this effect.

Reply:
> It is only possible to override this setting at the cluster level since we'd want all indices to have this effect.

In that case, why not change to below:

```
this.setMaxClauseCount(INDICES_MAX_CLAUSE_COUNT_SETTING.get(settings));
clusterService.getClusterSettings().addSettingsUpdateConsumer(INDICES_MAX_CLAUSE_COUNT_SETTING, this::setMaxClauseCount);
```

```
private void setMaxClauseCount(int maxClauseCount) {
  this.maxClauseCount = maxClauseCount;
  IndexSearcher.setMaxClauseCount(maxClauseCount);
}

// Used in SearchModule above
public int getMaxClauseCount() {
  return this.maxClauseCount;
}
```

Reply:
I made it a static method and now using getter and setter.